homeassistant/components/tts/entity.py

```diff
@@ -89,6 +89,14 @@ def default_options(self) -> Mapping[str, Any] | None:
         """Return a mapping with the default options."""
         return self._attr_default_options
 
+    @cached_property
+    def supports_streaming_input(self) -> bool:
+        """Return if the TTS engine supports streaming input."""
+        return (
+            TextToSpeechEntity.async_stream_tts_audio.__code__
```

Comment:
This makes logical sense, but it seems like a bad idea to depend on the code difference here. Shouldn't we just have a property that entity authors override?

Reply:
Updated to make it mimic `async_supports_options_flow`